### ISSUE TITLE
Mutable collection as a property for custom-elements

### ISSUE THREAD

**Issue by pH200** [staltz/cycle#106]
```
Cycle.registerCustomElement('x-custom-element', customElement);
var theArray = [];
h('x-custom-element', { array: theArray });
```

And pushing items into the array would not cause the notifications to the custom element. Which was likely work as expected, because it's the same old instance for the array object. So instead of pushing items we need to create a new instance of the array for the property for causing notification.

However, this is not reasonable for arrays which have large collection and updates frequently. Therefore, we need a way for custom elements raising notifications for the mutable arrays.

**Comment by pH200** [comment]
Example:
https://gist.github.com/pH200/6a3afc687195d3194369

**Comment by staltz** [comment]
Solved in v0.20.3
